### PR TITLE
[Feature] - Find points of interest

### PR DESCRIPTION
**Pull request by sausti40-ford**
Adds the ability to use the poi v3 api for finding regular points of interest. You can use the categories parameter to specifically search for fuel.

Some refactoring of RequestBuilder to ensure that the builder itself determines what the type of the response is - each "RequestBuilder" now has their own type. Minor re-organisation and removal of type prefixes.

### CODE REVIEW

File: FordPointsOfInterestService/Sources/Models/FuelStation.swift [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    public let searchString: String?
+    public let locale: String
+    public let numberOfRecords: Int?
+    public let categories: [String]
+    public let searchLocation: POICoordinate
+
+    public init(searchString: String? = nil,
+                locale: Locale = Locale.current,
+                numberOfRecords: Int? = nil,
+                categories: [String] = ["FUEL"],
```
Comment (mferra34-ford) [inline FordPointsOfInterestService/Sources/Models/FuelStation.swift]: Maybe I'm overthinking it, but if we share the module with FP, we have to make sure here no one needs to search for two categories at the same time. But right now, I don't see an issue

File: FordPointsOfInterestService/Sources/Models/Category.swift [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    public init(rawValue: String) {
+        self.rawValue = rawValue
+    }
+
+    public static let food = Category(rawValue: "FOOD")
+    public static let coffee = Category(rawValue: "COFFEE")
+    public static let entertainment = Category(rawValue: "ENTERTAINMENT")
+    public static let travel = Category(rawValue: "TRAVEL")
+    public static let money = Category(rawValue: "MONEY")
+    public static let community = Category(rawValue: "COMMUNITY")
```
Comment (mferra34-ford) [inline FordPointsOfInterestService/Sources/Models/Category.swift]: is it an exhausting list?

File: FordPointsOfInterestService/Sources/Parameters/PointOfInterestParameters.swift [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+//  Copyright © 2020 DevPair26. All rights reserved.
+//
+
+import CoreLocation
+
+public struct PointOfInterestParameters: Codable {
+    public let searchString: String?
+    public let locale: String
+    public let numberOfRecords: Int?
+    public let categories: Set<Category>
```
Comment (mferra34-ford) [inline FordPointsOfInterestService/Sources/Parameters/PointOfInterestParameters.swift]: Set?

File: FordPointsOfInterestService/Sources/Models/Category.swift (hunk 2 above)
Comment (sausti40-ford) [reply FordPointsOfInterestService/Sources/Models/Category.swift]: I am unsure. I only added the ones that definitely worked with /v3/, some of the old ones don't work.
The setup of this is done in a non-exhaustive way, as `Category` can be initialized with any rawValue

**Pull request merged**